internal/config/from_env.go

```diff
@@ -56,7 +56,7 @@ func (c *envConfig) GetWithSource(hostname, key string) (string, string, error)
 func (c *envConfig) CheckWriteable(hostname, key string) error {
 	if hostname != "" && key == "oauth_token" {
 		if token, env := AuthTokenFromEnv(hostname); token != "" {
-			return fmt.Errorf("read-only token in %s cannot be modified", env)
+			return fmt.Errorf("The '%s' environment variable is not empty, please clear it to save authentication data", env)
```

Comment:
It's an improvement, but I'm not 100% sure either about this new error message. It makes it sound like setting GITHUB_TOKEN or GH_TOKEN in your environment is something wrong that you should clear to proceed.

Since setting environment variables is a perfectly valid way to authenticate, maybe the error message shouldn't instruct users to clear the environment variable, but should instruct them to change the token that's saved in the variable if they need to re-authenticate?

Reply:
Options:
1. ``This command is not compatible with pre-existing environment variable '%s'. You will need to confirm & update the variable yourself as needed.``
2. ``Pre-existing login token '%s' recognized. If you want this application to manage your login, clear the environment variable.``
3. ``Pre-existing login token '%s' recognized. Re-authentication will require changing the environment variable manually.``

Reply:
Thank you for the suggestions. Upon further deliberation, I decided that the `config` package isn't the right place to print an error message about environment variables being read-only. Instead, that responsibility should fall on `auth` commands themselves. I've opened an alternate PR here #2813